docs/source/tutorials_source/platform/tutorial_active_learning_detectron2.py

```diff
@@ -0,0 +1,340 @@
+# each sample and for each scoring method.
+scorer = ScorerObjectDetection(obj_detection_outputs)
+scores = scorer.calculate_scores()
+# %% 
+# Let's have a look at the sample with the highest
+# uncertainty_margin score.
+#
+# .. note::
+#    A high uncertainty margin means that the image contains at least one 
+#    bounding box for which the model is unsure about the class of the object 
+#    in the bounding box. Read more about how our active learning scores are 
+#    calculated here:
+#    :py:class:`lightly.active_learning.scorers.detection.ScorerDetection`
+max_score = scores['uncertainty_margin'].max()
+idx = scores['uncertainty_margin'].argmax()
+print(f'Highest uncertainty_margin score found for idx {idx}: {max_score}')
+
+# %%
+# Let's have a look at this particular image and show the model 
+# prediction for it.
+fname = os.path.join(DATASET_ROOT, al_agent.unlabeled_set[idx])
+predict_and_overlay(predictor, fname)
+
+# %%
+# Finally, we can tell our agent to select the top 100 images to annotate and
+# improve our existing model. We pick the sampling method called `CORAL` which
+# is a combination of Coreset and Active Learning. Whereas Coreset maximizes
+# the image diversity based on the embeddings, active learning aims at selecting
+# images where our models struggles the most. 
+config = SamplerConfig(
+  n_samples=100, 
+  method=SamplingMethod.CORAL, 
+  name='active-learning-loop-1'
+)
+al_agent.query(config, scorer)
+
+# %%
+# We can access the newly added data from the agent.
+print(len(al_agent.added_set))
+
+# %%
+# Let's have a look at the first 5 entries.
+print(al_agent.added_set[:5])
+
+# %%
+# Let's show model predictions for the first 5 images.
+to_label = [os.path.join(DATASET_ROOT, x) for x in al_agent.added_set]
+for i in range(5):
+  predict_and_overlay(predictor, to_label[i])
+
+# %%
+# What's Next?
+# -------------
+# 
+# We showed in this tutorial how you can use Lightly Active Learing to discover 
+# the images you should label next. You can close the loop by annotating 
+# the 100 images and re-train your model. Then start the next iteration 
+# by making new model predictions on the updated unlabeled_set (which will now 
+# contain 100 fewer images).
+#
+# Using Lightly Active Learning has two advantages:
+#
+# - By letting the model chose the next batch of images to label we achieve 
+#   a higher accuracy faster. We're only labeling the images having a great impact.
+# 
+# - By combining the model predictions with the image embeddings we make sure we 
+#   don't select many similar images. Imagine the model being very bad at small 
+#   red cars and the 100 images therefore would only contain this set of images. 
+#   We might overfit the model because it suddenly has too many training examples 
+#   of small red cars.
+
+# %%
+# After retraining our model on the newly labeled 100 images 
+# we can do another active learning iteration with the updated unlabeld set.
+print(len(al_agent.unlabeled_set))
```

Comment:
Same here.

Maybe writing too little text here is a mistake. It could be worth to elaborate more. For example:

At any time during the active learning process, the active learning agent provides access to all images in the active learning pool, the unlabeled images, the labeled images, and the images which were selected in the last active learning query:

```Python
# get the list of filenames of all samples in your pool
al_agent.query_set

# get the list of filenames of all unlabeled samples
# (this is the query_set without the labeled_set)
al_agent.unlabeled_set

# get the list of filenames of all labeled_samples:
# (this is the query_set without the unlabeled_set)
al_agent.labeled_set

# get the list of filenames of all samples selected in the last query
# this is a subset of the labeled_set
al_agent.added_set
```

Reply:
I did not want to include this because it adds a lot of complexity. I'll think about a way to simplify.